src/python/dxpy/bindings/dxglobalworkflow.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (C) 2013-2018 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+"""
+DXGlobalWorkflow Handler
+++++++++++++++++++++++++
+
+"""
+
+from __future__ import print_function, unicode_literals, division, absolute_import
+
+import dxpy
+from . import DXObject, DXExecutable, DXJob, verify_string_dxid
+from ..exceptions import DXError
+from ..compat import basestring
+
+####################
+# DXGlobalWorkflow #
+####################
+
+class DXGlobalWorkflow(DXObject, DXExecutable):
+    '''
+    Remote global workflow object handler.
+
+    '''
+
+    _class = "globalworkflow"
+
+    def __init__(self, dxid=None, name=None, alias=None):
+        DXObject.__init__(self)
+        if dxid is not None or name is not None:
+            self.set_id(dxid=dxid, name=name, alias=alias)
+
+    def set_id(self, dxid=None, name=None, alias=None):
+        '''
+        :param dxid: Global workflow ID
+        :type dxid: string
+        :param name: Global workflow name
+        :type name: string
+        :param alias: Global workflow version or tag
+        :type alias: string
+        :raises: :exc:`~dxpy.exceptions.DXError` if *dxid* and some other input are both given or if neither *dxid* nor *name* are given
+
+        Discards the currently stored ID and associates the handler
+        with the requested parameters.  Note that if *dxid* is given,
+        the other fields should not be given, and if *name* is given,
+        *alias* has default value "default".
+
+        '''
+        self._dxid = None
+        self._name = None
+        self._alias = None
+        if dxid is not None:
+            if name is not None or alias is not None:
+                raise DXError("Did not expect name or alias to be given if dxid is given")
+            verify_string_dxid(dxid, self._class)
+            self._dxid = dxid
+        elif name is not None:
+            self._name = name
+            if not isinstance(name, basestring):
+                raise DXError("Global workflow name needs to be a string: %r" % (name,))
+            if alias is not None:
+                if not isinstance(alias, basestring):
+                    raise DXError("Global workflow alias needs to be a string: %r" % (alias,))
+                self._alias = alias
+            else:
+                self._alias = 'default'
```

Comment:
May want to add an `else` clause here to handle the no-input case if it's not handled well otherwise.

Reply:
Thanks for the note. In this case we don't need an else since we won't be doing anything i.e. setting id, name, or alias if neither id or name values were provided in the arguments.